src/util/string_utils.c

```diff
@@ -0,0 +1,90 @@
+#include "string_utils.h"
+#include "dict.h"
+#include <string.h>
+#include "../redisai_memory.h"
+
+RedisModuleString *RAI_HoldString(RedisModuleCtx *ctx, RedisModuleString *str) {
```

Comment:
This util is a good idea! But the first argument is redundant (we always send NULL as ctx).

When I come to think of it, if we know that we won't use the auto_memory mechanism, we may consider adding a util function such as "RAI_FreeString(RedisModuleString *str)", and replace every call for RedisModule_FreeString with that - so we avoid the annoying bugs. This can probably be done in another PR...